### DISCUSSION TITLE
Is SpatiaLite required for valhalla_build_tiles and valhalla_run_route?

### DISCUSSION THREAD

**Discussion by mloskot** [valhalla/valhalla#3110] [Q&A] [answered]
Is the SpatiaLite a hard requirement for the **minimal** use case of Valhalla, that is, building of the library and the two tools: `valhalla_build_tiles` and `valhalla_run_route`?

Is it even feasible to make optional (the CMake configuration) and `#ifdef` its uses in the code?

I'm contemplating a PR, but I'm still unclear if it is a hard requirement :-)

**Comment by kevinkreiser** [comment, marked as answer]
@mloskot it is not required at all for `valhalla_run_route`

for `valhalla_build_tiles` technically we could change the code so that it wouldnt be required. spatialite is used to do point in polygon look ups for adding timezone and administrative associations to nodes in the graph. the timezone and admin polygons are stored in spatialite dbs. since both of those data sources are optional technically spatialite could be optoinal too but we would need to make a bunch of changes to a few parts of mjolnir to make that the case. i didnt scope it out to refresh my memory but i suspect it wouldnt be a trivial (in size) refactor (though i do think it would be pretty mechanical)

**Reply by mloskot** [reply]
Thank you very much for the explanation. I will see if I can dive into it, no promises though ☺️

**Comment by nilsnolde** [comment]
I'd like that, spatialite can be troublesome. You're probably aware of it and don't need it for your use case, just wanna leave this caution here: https://github.com/valhalla/valhalla/issues/2320 (there's a little bug with roundabout navigation if there's no admin DB), apart from no time-dependent routing, border crossing, streetside confusion etc
